### ISSUE TITLE
Missing 'boto3_session' in aws-data-wrangler/awswrangler/s3.py 

### ISSUE THREAD

**Issue by hjuhel-cdpq** [awslabs/aws-data-wrangler#172]
**Describe the bug**
Using ```s3.to_parquet```with mode ```overwrite_partitions``` fails since the call to ```delete_objects``` does not propagate the ```boto3_session```

**To Reproduce**
* Try to create a dataset while providing a boto3 session
```python
import awswrangler as wr
import pandas as pd
import boto3

# The custom boto3 session 
sess = boto3.Session(aws_access_key_id='<foo>', aws_secret_access_key='<bar>')

# The dataframe to write
df = pd.DataFrame({'one': [0,], 'two': ["spam"], 'three': [True]})

wr.s3.to_parquet(
    df=df,
    path='s3://path',
    dataset=True,
    mode="overwrite_partitions",
    partition_cols=["two"],
    boto3_session=sess
)
```

The previous snipet results in an authentication error, probably because of the missing boto3_session parameter in the delete call:

```python
if mode == "overwrite_partitions":
    delete_objects(path=prefix, use_threads=use_threads
```

**Comment by igorborgest** [comment]
Hi @hjuhel-cdpq! Thank you very much for report it.

Will be fixed and released later today in the version `1.0.2`.

**Comment by igorborgest** [comment]
Thanks @hjuhel-cdpq, this issue is resolved on version `1.0.2`. Please, give feedback when possible.
